src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -442,14 +453,15 @@ def get_outer_index(array):
     does not exist then raise an exception.
 
     :param array: the array being examined.
-    :type array: :py:class:`psyclone.psyir.nodes.ArrayReference`
+    :type array: :py:class:`psyclone.psyir.nodes.ArrayMixin`
 
     :returns: the outermost index of the array that is a Range node.
 
     :raises IndexError: if the array does not contain a Range node.
 
     '''
-    for child in reversed(array.children):
+    # return array.walk(Range)[-1]
+    for child in reversed(array.indices):
         if isinstance(child, Range):
             return child.position
```

Comment:
Is the fact that this position isn't relative to the `indices` OK?

Reply:
Yes, if the resulting index is used in a `node.children[index]` expression. But since the "index" here is an overloaded term (children index or dimension index) I changed the docstring to be clear about this.

Also, since this is a utility for arrays - not specific for this transformation (e.g. is imported for other transformations), I moved it to a method of the ArrayMixin.